pkg/virt-operator/resource/apply/delete.go

```diff
@@ -464,13 +465,55 @@ func DeleteAll(kv *v1.KubeVirt,
 		}
 	}
 
+	if err = deleteKubeVirtLabelsFromNodes(clientset); err != nil {
+		return err
+	}
+
 	err = deleteDummyWebhookValidators(kv, clientset, stores, expectations)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func deleteKubeVirtLabelsFromNodes(clientset kubecli.KubevirtClient) error {
+	nodeList, err := clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	for _, node := range nodeList.Items {
+		labels := node.GetLabels()
+		modified := false
+
+		if labels == nil {
+			continue
+		}
+		patchSet := patch.New()
+		for labelkey := range labels {
+			if strings.HasPrefix(labelkey, "kubevirt.io/") {
+				patchSet.AddOption(patch.WithRemove(fmt.Sprintf("/metadata/labels/%s", patch.EscapeJSONPointer(labelkey))))
+				modified = true
+			}
+		}
+
+		if !modified {
```

Comment:
Instead of using modified you can simply check if the patchSet is empty, there is a function for that